artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+import concurrent
+import inspect
+import logging
+import os
+import typing
+
+from google.cloud import bigquery
+
+from artcommonlib import logutil
+from artcommonlib.konflux import konflux_build_record
+
+
+class KonfluxDb:
+    REQUIRED_VARS = ['GOOGLE_CLOUD_PROJECT', 'GOOGLE_APPLICATION_CREDENTIALS', 'DATASET_ID', 'TABLE_ID']
+
+    def __init__(self):
+        self._check_env_vars()
+        self.client = bigquery.Client()
+        self.dataset_id = os.environ['DATASET_ID']
+        self.table_ref = f'{self.client.project}.{self.dataset_id}.{os.environ["TABLE_ID"]}'
+        self.logger = logutil.get_logger(__name__)
+        self.column_names = self._get_column_names()  # e.g. "(name, group, version, release, ... )"
+
+        # Make the gcp logger less noisy
+        logging.getLogger('google.auth.transport.requests').setLevel(logging.WARNING)
+
+    def _check_env_vars(self):
+        """
+        Check if all required env vars are defined. Raise an exception otherwise
+        """
+
+        for var in self.REQUIRED_VARS:
+            if not os.getenv(var):
+                raise RuntimeError(f'Environment variable {var} must be defined')
+
+    @staticmethod
+    def _get_column_names() -> str:
+        """
+        Inspects the KonfluxBuildRecord class definition, and returns its fields a list
+        To be used with INSERT statements
+        """
+
+        fields = []
+        parameters = inspect.signature(konflux_build_record.KonfluxBuildRecord.__init__).parameters
+
+        for param_name, _ in parameters.items():
+            if param_name == 'self':
+                continue
+            fields.append(f"`{param_name}`")
+
+        return f'({", ".join(fields)})'
+
+    def add_build(self, build: konflux_build_record.KonfluxBuildRecord):
+        """
+        Insert a build record into Konflux DB
+        """
+
+        query = f'INSERT INTO `{self.table_ref}` {self.column_names} VALUES '
```

Comment:
If possible, it is better to use parameterized queries to avoid SQL injection and make the code more readable. But I am ok with the current implementation for now.

https://cloud.google.com/bigquery/docs/parameterized-queries

Reply:
I think it's doable, but needs time and care to implement in that `INSERT` statement. Doozer with ART's service account is the only actor who will be able to execute such statements, so I would not worry about SQL injection. Same applies for `SELECT` statements executed via the API (only Elliott with our service account will use that). We can still track this as a future enhancement

Reply:
I have created [ART-10667](https://issues.redhat.com/browse/ART-10667) as a investigation follow-up